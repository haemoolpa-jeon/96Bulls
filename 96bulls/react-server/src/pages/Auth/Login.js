import React, { Component } from 'react';
import Form from '../components/common/Form';
import Input from '../components/common/Input';
import Button from '../components/common/Button';
import ErrorMessage from '../components/common/ErrorMessage';
import firebase from '../../config/firebase';
import styles from './Login.module.scss';
import { Link } from 'react-router-dom';


class Login extends Component {
  state = {
    email: '',
    password: '',
    errors: [],
    usersRef: firebase.database().ref('users'),
    loading: false,
  };

  isFormValid = () => {
    const errors = [];
    let error;

    if (this.isFormEmpty(this.state)) {
      error = { message: 'Please fill in all the field!' };
      this.setState({ errors: errors.concat(error), loading: false });
      return false;
    }
    return true;
  };

  isFormEmpty = ({
    email, password,
  }) => !email.length || !password.length;

  displayErrors = errors => errors.map((error, i) => <p key={i}>{error.message}</p>);

  handleChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    this.setState({ errors: [], loading: true }, () => {
      const { email, password, errors } = this.state;
      if (this.isFormValid(this.state)) {
        firebase
          .auth()
          .signInWithEmailAndPassword(email, password)
          .then((signedInUser) => {
            console.log(signedInUser);
          })
          .catch((err) => {
            console.error(err);
            if (err.code === 'auth/wrong-password') {
              const error = { message: 'Invalid Password' };
              this.setState({
                errors: errors.concat(error), loading: false,
              });
            }
            if (err.code === 'auth/user-not-found') {
              const error = { message: 'User not found' };
              this.setState({
                errors: errors.concat(error), loading: false,
              });
            }
            if (err.code === 'auth/invalid-email') {
              const error = { message: 'Please check your email' };
              this.setState({
                errors: errors.concat(error), loading: false,
              });
            }
          });
      }
    });
  };

  render() {
    const {
      email, password, errors, loading,
    } = this.state;

    return (
      <section>
        <Form className={styles['login-form']}>
          <h1>ALTIS ACADEMIA</h1>
          <Input
            id="email"
            value={email}
            name="email"
            placeholder="E-mail"
            type="email"
            onChange={this.handleChange}
          />
          <Input
            id="password"
            value={password}
            name="password"
            placeholder="Password"
            type="password"
            onChange={this.handleChange}
          />
          {errors.length > 0 && (
            <ErrorMessage>
              {this.displayErrors(errors)}
            </ErrorMessage>
          )}
          <p className={styles['login-info']}>
            {'Is this your first time? '}
            <Link to="/register">Sign Up</Link>
          </p>
          <div className={styles['button-wrapper']}>
            <Button
              disabled={loading}
              loader={loading}
              onClick={this.handleSubmit}
            >
            Log In
            </Button>
          </div>
        </Form>
      </section>
    );
  }
}

export default Login;
