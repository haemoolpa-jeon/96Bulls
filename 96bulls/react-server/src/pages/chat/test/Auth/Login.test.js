import React from 'react';
import Login from '../../components/Auth/Login';
import {shallow} from 'enzyme/build';

describe('Login', () => {
  let component = null;
  it('초기 렌더링이 문제없이 되야함', () => {
    component = shallow(<Login/>);
  });

  it('초기 렌더링 스냅샷 일치함', () => {
    expect(component).toMatchSnapshot();
  });
});
