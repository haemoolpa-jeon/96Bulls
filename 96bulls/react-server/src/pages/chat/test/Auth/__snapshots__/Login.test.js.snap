// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login 초기 렌더링 스냅샷 일치함 1`] = `
<section>
  <Form
    className="login-form"
  >
    <h1>
      OPEN CHAT
    </h1>
    <Input
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="이메일"
      type="email"
      value=""
    />
    <Input
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="비밀번호(최소 6자)"
      type="password"
      value=""
    />
    <p
      className="login-info"
    >
      처음 방문하셨나요? 
      <Link
        replace={false}
        to="/register"
      >
        회원가입
      </Link>
    </p>
    <div
      className="button-wrapper"
    >
      <Button
        disabled={false}
        loader={false}
        onClick={[Function]}
      >
        로그인
      </Button>
    </div>
  </Form>
</section>
`;
